crates/turbo-updater/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::Duration;
+
+use colored::*;
+use serde::Deserialize;
+use thiserror::Error as ThisError;
+use update_informer::{Check, Package, Registry, Result as UpdateResult};
+
+mod ui;
+
+const DEFAULT_TIMEOUT: Duration = Duration::from_millis(800);
+const DEFAULT_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24);
+
+#[derive(ThisError, Debug)]
+pub enum UpdateNotifierError {
+    #[error("Failed to write to terminal")]
+    RenderError(#[from] ui::utils::GetDisplayLengthError),
+}
+
+#[derive(Deserialize)]
+struct NpmVersionData {
+    version: String,
+}
+
+struct NPMRegistry;
+
+impl Registry for NPMRegistry {
+    const NAME: &'static str = "npm_registry";
+    fn get_latest_version(pkg: &Package, _timeout: Duration) -> UpdateResult<Option<String>> {
+        let url = format!("https://registry.npmjs.org/{}/latest", pkg);
```

Comment:
This URL should probably be: `<package-manager> config get registry`

Works for:
- yarn (https://registry.yarnpkg.com)
- npm
- pnpm

Breaks for:
- berry

Reply:
Discussed this and we're going to route to an API route we can control for per reasons.